client/src/ui/components/Button.tsx

```diff
@@ -1,11 +1,13 @@
 import React from 'react';
 import cx from 'classnames';
+import { Spinner } from './Spinner';
 
 interface ButtonProps extends React.HTMLProps<HTMLButtonElement> {
   iconStart?: React.ReactNode;
   iconEnd?: React.ReactNode;
   color: 'red' | 'grey';
   variant?: 'solid' | 'outline';
+  isLoading?: boolean;
```

Comment:
Wondering if it makes sense to set `disabled` to true automatically if `isLoading` is true

Reply:
@pradel this is indeed a very good idea. Updated and refactored where it was necessary.